### PR TITLE
Java 21 and spring boot migration code changes

### PR DESCRIPTION
**Pull request by dgurjar-ch**
Java 21 and spring boot migration code changes

JIRA Ticket Number_: CC-1059

### Type of change

* [x] New feature
* [x] Breaking change

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -217,18 +275,6 @@
                         <artifactId>maven-jar-plugin</artifactId>
                         <version>${maven-jar-plugin.version}</version>
                     </plugin>
-                    <plugin>
-                        <groupId>org.apache.maven.plugins</groupId>
-                        <artifactId>maven-surefire-plugin</artifactId>
```
Comment (dhawes-ch) [inline pom.xml:222]: Are unit tests definitely running ok with surefire having been removed?

File: pom.xml [hunk 2]
```diff
@@ -253,11 +299,17 @@
...
+                        <version>${jib-maven-plugin.version}</version>
                         <configuration>
-                          <to>
-                            <image>169942020521.dkr.ecr.eu-west-1.amazonaws.com/local/document-generator-consumer</image>
-                          </to>
+                            <from>
+                                <image>416670754337.dkr.ecr.eu-west-2.amazonaws.com/ci-corretto-build-21:latest</image>
+                            </from>
+                            <to>
+                                <image>416670754337.dkr.ecr.eu-west-2.amazonaws.com/local/document-generator-consumer:latest</image>
```
Comment (dhawes-ch) [inline pom.xml:308]: Believe this shouldn't have /local/ in the path

File: pom.xml [hunk 3]
```diff
@@ -25,58 +25,81 @@
...
 
         <!-- JUnit Testing -->
-        <junit-jupiter-engine.version>5.2.0</junit-jupiter-engine.version>
-        <mockito-junit-jupiter.version>2.18.0</mockito-junit-jupiter.version>
+        <junit-jupiter-engine.version>5.11.0</junit-jupiter-engine.version>
+        <mockito-junit-jupiter.version>5.12.0</mockito-junit-jupiter.version>
 
         <!-- Structured logging -->
-        <structured-logging.version>1.4.0-rc2</structured-logging.version>
+        <structured-logging.version>3.0.12</structured-logging.version>
```
Comment (dhawes-ch) [inline pom.xml:41]: Latest is 3.0.20.pom

File: src/main/resources/application.properties
Comment (dhawes-ch) [inline src/main/resources/application.properties:1]: Better to add `management.endpoints.enabled-by-default=false` here and enable what is required (looks like just health)

Comment (zrahman-ch) [reply src/main/resources/application.properties:1]: You can add this: `management.endpoints.web.exposure.include=health` and including the above.

File: pom.xml (hunk 1 above)
Comment (dgurjar-ch) [reply pom.xml:222]: yes , that's why I removed , But can add again

File: pom.xml (hunk 2 above)
Comment (dgurjar-ch) [reply pom.xml:308]: yes

File: pom.xml (hunk 1 above)
Comment (zrahman-ch) [reply pom.xml:222]: pipeline should run when a PR is raised on feature/CC-1059-java-21-migration but are you getting the surefire reports?

**Pull request merged**